### PR TITLE
Add centOS support to autoinstaller

### PR DESCRIPTION
**Pull request by theoddball**
Also made autorun auto update when starting

### CODE REVIEW

**Comment by samdivaio** [comment]
Check your PR. https://github.com/theoddball/NadekoBot-BashScript/pull/3
Fix Conflicts and Merge, then test for unix issues.

**Comment by theoddball** [comment]
In process of testing rn. Will return for updates

**Comment by theoddball** [comment]
Tested fine on centos.

**Comment by samdivaio** [comment]
Kwoth you can merge this now, its tested and working, I checked with CentOS 5,6,7 and Debian 7,8. @theoddball  took care of CentOS + minimals + Ubuntu 17.

Changes Includes:
- CentOS support added. (@theoddball)
- Screen clear on incorrect input. (ARU)
- CentOS added to `linuxAIO.sh`
- Auto Installer now checks for OS versions for Debian and CentOS before installing.
- Fixed changes from apt to apt-get. (minimal linux thing.)
- CentOS no longer require user confirmation to .NET prerequisites installation.

powered by Gwen 1.1

**Comment by theoddball** [comment]
added fedora, tested works

**Comment by samdivaio** [comment]
Need to add few other things I guess...
```
[root@fedora-512mb-nyc3-01 ~]# bash nadekoautoinstaller.sh
Welcome to NadekoBot Auto Prerequisites Installer. 
Would you like to continue? 
Your OS: Fedora 
OS Version: 24 
Architecture: x64
[y/n]: y
Running NadekoBot Auto-Installer

Preparing...
Yum command has been deprecated, redirecting to '/usr/bin/dnf --obsoletes --exclude=kernel* update -y'.
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

No such command: --obsoletes. Please use /usr/bin/dnf --help
It could be a DNF plugin command, try: "dnf install 'dnf-command(--obsoletes)'"
Yum command has been deprecated, redirecting to '/usr/bin/dnf install sudo -y'.
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

Last metadata expiration check: 0:01:27 ago on Thu Feb  9 02:16:35 2017.
Package sudo-1.8.16-3.fc24.x86_64 is already installed, skipping.
Dependencies resolved.
====================================================================== Package    Arch         Version                  Repository     Size
======================================================================Skipping packages with conflicts:
(add '--best --allowerasing' to command line to force their upgrade):
 sudo       x86_64       1.8.18p1-1.fc24          updates       724 k

Transaction Summary
======================================================================Skip  1 Package

Nothing to do.
Complete!
Yum command has been deprecated, redirecting to '/usr/bin/dnf install libunwind libicu -y'.
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

Last metadata expiration check: 0:01:29 ago on Thu Feb  9 02:16:35 2017.
Dependencies resolved.
====================================================================== Package         Arch         Version             Repository     Size
======================================================================Installing:
 libicu          x86_64       56.1-7.fc24         updates       8.3 M
 libunwind       x86_64       1.1-11.fc24         fedora         64 k

Transaction Summary
======================================================================Install  2 Packages

Total download size: 8.3 M
Installed size: 29 M
Downloading Packages:
(1/2): libunwind-1.1-11.fc24.x86_64.r 240 kB/s |  64 kB     00:00    
(2/2): libicu-56.1-7.fc24.x86_64.rpm   15 MB/s | 8.3 MB     00:00    
----------------------------------------------------------------------Total                                 4.2 MB/s | 8.3 MB     00:01     Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Installing  : libicu-56.1-7.fc24.x86_64                         1/2   Installing  : libunwind-1.1-11.fc24.x86_64                      2/2   Verifying   : libunwind-1.1-11.fc24.x86_64                      1/2   Verifying   : libicu-56.1-7.fc24.x86_64                         2/2 
Installed:
  libicu.x86_64 56.1-7.fc24        libunwind.x86_64 1.1-11.fc24       
Complete!
Yum command has been deprecated, redirecting to '/usr/bin/dnf -y install http://li.nux.ro/download/nux/dextop/el7/x86_64/nux-dextop-release-0-5.el7.nux.noarch.rpm epel-release'.
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

Last metadata expiration check: 0:01:43 ago on Thu Feb  9 02:16:35 2017.
No package epel-release available.
Error: Unable to find a match.
Yum command has been deprecated, redirecting to '/usr/bin/dnf install git opus opus-devel ffempeg tmux -y'.
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

Last metadata expiration check: 0:01:45 ago on Thu Feb  9 02:16:35 2017.
No package ffempeg available.
Error: Unable to find a match.

NadekoBot Prerequisites Installation completed...
 ```

**Comment by theoddball** [comment]
Gotcha
Will change to dnf or you can, I'm going sleep

**Comment by samdivaio** [comment]
Fedora is messed up, I need to get bunch of stuffs to just install ffmpeg, current changes, we need few more... https://github.com/samdivaio/NadekoBot-BashScript/commit/4dced4e5400c45680b546164af16a935325c79a3 changes. Will check on this with 23 aswell. Currently procrastinating.

**Comment by samdivaio** [comment]
@theoddball Revert Fedora changes and let @Kwoth merge the centOS + other changes. **Save the changes you have right now in a different branch before doing that.**

**Comment by theoddball** [comment]
done go for merge

File: nadekoautoinstaller.sh [hunk 1]
```diff
@@ -163,26 +166,51 @@ read -n 1 -s -p "Press any key to continue..."
...
+		exit 1
+	fi
 fi
 
 echo
-echo "Installation completed..."
+echo "NadekoBot Prerequisites Installation completed..."
 sleep 2
 
 rm nadekoautoinstaller.sh && bash linuxAIO.sh
```
Comment (samdivaio) [inline nadekoautoinstaller.sh]: Gwen can you please remove `bash linuxAIO.sh` from this line. It doesnt have to trigger it twice as its already running from AIO, shutting down will only double it. Thanks <3

**Comment by theoddball** [comment]
Done

**Comment by Kwoth** [comment]
@samdivaio ready?

**Comment by samdivaio** [comment]
Good to go 👍

**Pull request merged**